examples/docs-examples/examples/working-with-results/multi-command-endstream.mjs

```diff
@@ -6,17 +6,23 @@ import { TaskExecutor } from "@golem-sdk/golem-js";
     yagnaOptions: { apiKey: "try_golem" },
   });
 
-  const result = await executor.run(async (ctx) => {
-    const res = await ctx
-      .beginBatch()
-      .uploadFile("./worker.mjs", "/golem/input/worker.mjs")
-      .run("node /golem/input/worker.mjs > /golem/input/output.txt")
-      .run("cat /golem/input/output.txt")
-      .downloadFile("/golem/input/output.txt", "./output.txt")
-      .endStream();
+  try {
+    const result = await executor.run(async (ctx) => {
+      const res = await ctx
+        .beginBatch()
+        .uploadFile("./worker.mjs", "/golem/input/worker.mjs")
+        .run("node /golem/input/worker.mjs > /golem/input/output.txt")
+        .run("cat /golem/input/output.txt")
+        .downloadFile("/golem/input/output.txt", "./output.txt")
+        .endStream();
 
-    res.on("data", (result) => console.log(result));
-    res.on("error", (error) => console.error(error));
-    res.on("close", () => executor.end());
-  });
+      res.on("data", (result) => console.log(result));
+      res.on("error", (error) => console.error(error));
+      res.on("close", () => {});
```

Comment:
I wouldn't change that. The original code was waiting here for the stream to be "closed".

Alternative would be to something like `return new Promise((resolve) => res.once("close", resolve))` instead and then the whole `executor.run` would really wait for this particular event.